src/routes/DetailsPage.svelte

```diff
@@ -23,30 +25,33 @@
     hasError: false,
   };
 
-  onMount(async () => {
-    weatherData.isLoading = true;
+  $: temperatureSign = temperatureUnitSymbols[$temperatureUnit];
+  $: temperatureUnit.subscribe(() => {
+    fetchWeather();
+  });
 
+  const fetchWeather = async () => {
+    weatherData.isLoading = true;
     const { date, latitude: lat, longitude: lng } = params;
-
     const response = await weatherService.getHourlyForecast(lat, lng, date);
     hourlyData = response.map(data => ({
       ...data,
       time: DateTime.fromISO(data.time).toFormat('HH:mm'),
     }));
-
     if (!hourlyData.length) {
       weatherData.isLoading = false;
       weatherData.hasError = true;
 
       return;
     }
-
     hourlyData = hourlyData.map((el, idx) => ({
       ...el,
       dayPeriod: getDayPeriodByDate({ date: response[idx].time, lat, lng }),
     }));
-
     weatherData.isLoading = false;
+  };
+  onMount(async () => {
```

Comment:
in this case we have 2 similar request on mount: 
![image](https://github.com/user-attachments/assets/bba168a6-521f-43b0-b8ea-256baf7d288d)
can we get rid of this?

Reply:
I'll try)